javascript/modules/client/src/client-erc20.ts

```diff
@@ -241,3 +260,79 @@ function unwrapWithdrawParams(
     params.reference ?? "",
   ];
 }
+
+/// HELPER FUNCTIONS TODO: move to utils?
+function setProposalStatus(proposals: IErc20VotingProposal[]) {
```

Comment:
I would:
- call it `getProposalsWithStatus`
- Do a copy of the input data (`const result = []`), then at each iteration `push` an `Object.assign({}, proposal, {status: ...})`  to the results array
- Return the new array